packages/react-data-grid/src/utils/viewportUtils.ts

```diff
@@ -58,84 +27,77 @@ function getColumnCountForWidth<R>(columns: CalculatedColumn<R>[], initialWidth:
   return count;
 }
 
-export function getNonFrozenVisibleColStartIdx<R>(columns: CalculatedColumn<R>[], scrollLeft: number): number {
-  let remainingScroll = scrollLeft;
-  const lastFrozenColumnIndex = findLastFrozenColumnIndex(columns);
-  const nonFrozenColumns = columns.slice(lastFrozenColumnIndex + 1);
-  let columnIndex = lastFrozenColumnIndex;
-  while (remainingScroll >= 0 && columnIndex < nonFrozenColumns.length) {
-    columnIndex++;
-    const column = columns[columnIndex];
-    remainingScroll -= column ? column.width : 0;
-  }
-  return max(columnIndex, 0);
-}
-
-export function getNonFrozenRenderedColumnCount<R>(columnMetrics: ColumnMetrics<R>, viewportDomWidth: number, scrollLeft: number): number {
-  const { columns, totalColumnWidth } = columnMetrics;
-  if (columns.length === 0) {
-    return 0;
-  }
-  const colVisibleStartIdx = getNonFrozenVisibleColStartIdx(columns, scrollLeft);
-  const totalFrozenColumnWidth = getTotalFrozenColumnWidth(columns);
-  const viewportWidth = viewportDomWidth > 0 ? viewportDomWidth : totalColumnWidth;
-  const firstColumn = columns[colVisibleStartIdx];
-  // calculate the portion width of first column hidden behind frozen columns
-  const scrolledFrozenWidth = totalFrozenColumnWidth + scrollLeft;
-  const firstColumnHiddenWidth = scrolledFrozenWidth > firstColumn.left ? scrolledFrozenWidth - firstColumn.left : 0;
-  const initialWidth = viewportWidth - totalFrozenColumnWidth + firstColumnHiddenWidth;
-  return getColumnCountForWidth(columns, initialWidth, colVisibleStartIdx);
-}
-
-export interface VisibleBoundaries {
-  rowVisibleStartIdx: number;
-  rowVisibleEndIdx: number;
+interface VerticalRangeToRenderParams {
+  height: number;
+  rowHeight: number;
+  scrollTop: number;
+  rowsCount: number;
+  scrollDirection: SCROLL_DIRECTION;
+  overscanRowCount?: number;
 }
 
-export function getVisibleBoundaries(gridHeight: number, rowHeight: number, scrollTop: number, rowsCount: number): VisibleBoundaries {
-  const renderedRowsCount = ceil(gridHeight / rowHeight);
+export function getVerticalRangeToRender({
+  height,
+  rowHeight,
+  scrollTop,
+  rowsCount,
+  scrollDirection,
+  overscanRowCount = 2
+}: VerticalRangeToRenderParams) {
```

Comment:
I believe it's better to add return type, so that when we edit something wrong inside this component, we would get notified right away rather than in another file like `xxx is not in the return type`.

Reply:
Done